lib/setopt.c

```diff
@@ -77,6 +83,46 @@ CURLcode Curl_setstropt(char **charp, const char *s)
   return CURLE_OK;
 }
 
+CURLcode Curl_setblobopt(char **charp, const char *sblob)
+{
+  /* Release the previous storage at `charp' and replace by a dynamic storage
+     copy of `sblob'.
+     Return CURLE_OK, CURLE_BAD_FUNCTION_ARGUMENT or CURLE_OUT_OF_MEMORY. */
+
+  Curl_safefree(*charp);
+
+  if(sblob) {
+    size_t len = strlen(sblob);
+    struct curl_blob* st_blob = (struct curl_blob*)sblob;
+    if((len == CURL_BLOB_MAGIC_STRLEN) &&
+       (!memcmp(sblob, CURL_BLOB_MAGIC, CURL_BLOB_MAGIC_SIZE))) {
+        /* sblob contain struct curl_blob data, created with
+         * curl_init_blob_*. This is not ordinary string but a
+         * blob memory block */
+
+        bool dup = (st_blob->flags) & CURL_BLOB_DUPFLAG_COPYMASK;
+        struct curl_blob* st_blob_dup;
+        st_blob_dup = (struct curl_blob*)malloc(sizeof(struct curl_blob) +
+                                                (dup ? st_blob->len : 0));
```

Comment:
Maybe just use `Curl_memdup()` that we already have?

Reply:
No, because if we set dup, we don't copy the certificate data from same location